google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -1231,6 +1234,132 @@ public ObjectAccessControl patchAcl(ObjectAccessControl acl) {
     }
   }
 
+  @Override
+  public HmacKey createHmacKey(String serviceAccountEmail, Map<Option, ?> options) {
+    Span span = startSpan(HttpStorageRpcSpans.SPAN_NAME_CREATE_HMAC_KEY);
+    Scope scope = tracer.withSpan(span);
+    String projectId = Option.PROJECT_ID.getString(options);
+    if (projectId == null) {
+      projectId = this.options.getProjectId();
+    }
+    try {
+      return storage
+          .projects()
+          .hmacKeys()
+          .create(projectId, serviceAccountEmail)
+          .setUserProject(Option.USER_PROJECT.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      span.setStatus(Status.UNKNOWN.withDescription(ex.getMessage()));
+      throw translate(ex);
+    } finally {
+      scope.close();
+      span.end();
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<HmacKeyMetadata>> listHmacKeys(Map<Option, ?> options) {
+    Span span = startSpan(HttpStorageRpcSpans.SPAN_NAME_LIST_HMAC_KEYS);
+    Scope scope = tracer.withSpan(span);
+    String projectId = Option.PROJECT_ID.getString(options);
+    if (projectId == null) {
+      projectId = this.options.getProjectId();
+    }
+    try {
+      HmacKeysMetadata hmacKeysMetadata =
+          storage
+              .projects()
+              .hmacKeys()
+              .list(projectId)
+              .setServiceAccountEmail(Option.SERVICE_ACCOUNT_EMAIL.getString(options))
+              .setPageToken(Option.PAGE_TOKEN.getString(options))
+              .setMaxResults(Option.MAX_RESULTS.getLong(options))
+              .setShowDeletedKeys(Option.SHOW_DELETED_KEYS.getBoolean(options))
+              .execute();
+      return Tuple.<String, Iterable<HmacKeyMetadata>>of(
+          hmacKeysMetadata.getNextPageToken(), hmacKeysMetadata.getItems());
+    } catch (IOException ex) {
+      span.setStatus(Status.UNKNOWN.withDescription(ex.getMessage()));
+      throw translate(ex);
+    } finally {
+      scope.close();
+      span.end();
+    }
+  }
+
+  @Override
+  public HmacKeyMetadata getHmacKey(String accessId, Map<Option, ?> options) {
```

Comment:
Silly question, but what if getHmacKey used HMACMetadata as well? Throwing it out there but okay if you're okay with moving forward as-is as well.

Reply:
I don't really see the benefit of that, it just seems confusing to me. I'd rather move forward with accessId